uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/rest/PortletPrefsRESTController.java

```diff
@@ -0,0 +1,679 @@
+            }
+
+            logger.error(e.getMessage(), e);
+            return new ResponseEntity<>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        } catch (Exception e) {
+
+            logger.error("threw error trying to retrieve composite preferences", e);
+            return new ResponseEntity<>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * Get uPortal Preferecnces for portlet/soffit/webcomponent. This method provides a REST
+     * interface for uPortal for obtaining only the entity preferences for the
+     * portlet/soffit/webcomponent
+     *
+     * <p>The path for this method is /prefs/getprefs/fname. The fname is a string representing the
+     * fname of the portlet/soffit/webcomponent to return the composite entity preferences. These
+     * are the per-user preferences for the given portlet/soffit/webcomponent. Includes only the
+     * user set preferences, does not include any defaults set in the definition
+     */
+    /*maybe get rid of this one? don't know if there is any specific use for it given that the preferences needed might be set in
+     * any of the entity, definition, or descriptor preferences. But is useful if you only need the user set preferences and are
+     * not relying on a default in the definition.
+     */
+    @RequestMapping(value = "/prefs/getentityonlyprefs/{fname}", method = RequestMethod.GET)
+    public ResponseEntity getEntityOnlyPrefs(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            @ApiParam(value = "The portlet fname to find preferences for")
+                    @PathVariable(value = "fname")
+                    String fname) {
+        try {
+            IPortletDefinition portletDefinition = portletDao.getPortletDefinitionByFname(fname);
+            if (portletDefinition == null) {
+                return new ResponseEntity<>("ERROR: Portlet not found", HttpStatus.NOT_FOUND);
+            }
+            IPortletEntity entity =
+                    portletEntityRegistry.getOrCreateDefaultPortletEntity(
+                            request, portletDefinition.getPortletDefinitionId());
+
+            PortletPreferences defprefs =
+                    portletPreferencesFactory.createAPIPortletPreferences(
+                            request, entity, false, DEFINITION_MODE);
+            PortletPreferences entprefs =
+                    portletPreferencesFactory.createAPIPortletPreferences(
+                            request, entity, false, ENTITY_MODE);
+            Map<String, String[]> eprefs = getEntityOnly(entprefs.getMap(), defprefs.getMap());
+
+            return ResponseEntity.ok()
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .body(mapper.writeValueAsString(eprefs));
+        } catch (IllegalArgumentException e) {
+            if (e.getMessage().contains("No portlet definition found for id '")) {
```

Comment:
There isn't a better error ? because intercepting only on a message is never good. As example if there is a change on the text message this interception won't work anymore and nothing can permit to track it/keep the link to the case to intercept.

Reply:
There might be, but the error I'm catching is coming from the depths of uPortal itself, I didn't make that part. I'm not sure where it's coming from so I don't know where I'd even start to change it. If you know where that error is being generated I could look into it but I don't want to break any current functionality.